rados/ioctx.go

```diff
@@ -883,3 +883,77 @@ func (ioctx *IOContext) BreakLock(oid, name, client, cookie string) (int, error)
 		return int(ret), RadosError(int(ret))
 	}
 }
+
+// Snapshots
```

Comment:
It might be good to have the new functionality in a new "snapshot.go" file or similar.